### ISSUE TITLE
More problems installing and running Abelfunctions

### ISSUE THREAD

**Issue by harrybraden** [abelfunctions/abelfunctions#142]
Dear Chris, I am working with sage 7.5. The following looks like an earlier discussion, but I cant see the resolution of it. Help please?

```
sage: from abelfunctions import *
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-1-d165c0ae510e> in <module>()
----> 1 from abelfunctions import *

/Users/hwb/abelfunctions/abelfunctions/__init__.py in <module>()
     11 from abelfunctions.abelmap import AbelMap, Jacobian
     12 from abelfunctions.homology import symmetrize_periods
---> 13 from abelfunctions.puiseux import puiseux
     14 from abelfunctions.riemann_constant_vector import RiemannConstantVector
     15 from abelfunctions.riemann_surface import RiemannSurface

/Users/hwb/abelfunctions/abelfunctions/puiseux.py in <module>()
     41 import sympy
     42
---> 43 from abelfunctions.puiseux_series_ring import PuiseuxSeriesRing
     44
     45 from sage.all import gcd, xgcd

/Users/hwb/abelfunctions/abelfunctions/puiseux_series_ring.py in <module>()
     35 from sage.rings.all import IntegralDomain, CommutativeRing, Field
     36
---> 37 from abelfunctions.puiseux_series_ring_element import (
     38     is_PuiseuxSeries,
     39     PuiseuxSeries,

ImportError: No module named puiseux_series_ring_element
```

**Comment by cswiercz** [comment]
Hello Harry,

First, make sure you have the most recent version of Abelfunctions downloaded. Did you use the
```
$ sage setup.py install
```
method mentioned [here](https://github.com/abelfunctions/abelfunctions/blob/master/doc/Installation.md)? Also, make sure you navigate away from the `abelfunctions` directory after you install the package. Sage will first look for packages in the same directory as where you started Sage. The `setup.py` script puts the compiled output elsewhere so Sage will error when trying to import from the directory where you downloaded the software.

Also, pinging @chualynn who is helping maintain Abelfunctions after my departure.

**Comment by harrybraden** [comment]
Dear Chris,

I am installing via your second method as there is a problem with the first (I emailed the output in a previous email). Let me go through the steps as I am unclear on the directories
you describe in the instructions.

1, I have downloaded the latest .zip
2. I moved this to my top directory.
3. Unpacked this and renamed abelfunctions-master -> abelfunctions
4. Within this directory is LICENSE  notebooks spkg-check README.md doc	runtests.py spkg-install abelfunctions
5. Move to this directory, i.e. with these files in it and type 
    tar cjf abelfunctions.spkg  abelfunctions 
6. There is now abelfunctions.spkg in this directory
7. From the top directory run
    sage -p -c abelfunctions/abelfunctions.spkg
8. Get something that looks positive:

uccessfully installed abelfunctions
Package abelfunctions has no test suite.
Deleting temporary build directory
/Applications/SageMath-7.5.1.app/Contents/Resources/sage/local/var/tmp/sage/build/abelfunctions
Finished installing abelfunctions.spkg

Warning: it might be needed to update the Sage library before
installed packages work: you should run 'make' from $SAGE_ROOT
before running Sage.

9. No idea where make is to be run from, if at all.
10. Move to the directory abelfunctions and run sage (because I can’t manage to get sage to read a relative directory).
11. 
sage: from abelfunctions import *
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-1-d165c0ae510e> in <module>()
----> 1 from abelfunctions import *

/Users/hwb/abelfunctions/abelfunctions/__init__.py in <module>()
     11 from abelfunctions.abelmap import AbelMap, Jacobian
     12 from abelfunctions.homology import symmetrize_periods
---> 13 from abelfunctions.puiseux import puiseux
     14 from abelfunctions.riemann_constant_vector import RiemannConstantVector
     15 from abelfunctions.riemann_surface import RiemannSurface

/Users/hwb/abelfunctions/abelfunctions/puiseux.py in <module>()
     41 import sympy
     42
---> 43 from abelfunctions.puiseux_series_ring import PuiseuxSeriesRing
     44
     45 from sage.all import gcd, xgcd

/Users/hwb/abelfunctions/abelfunctions/puiseux_series_ring.py in <module>()
     35 from sage.rings.all import IntegralDomain, CommutativeRing, Field
     36
---> 37 from abelfunctions.puiseux_series_ring_element import (
     38     is_PuiseuxSeries,
     39     PuiseuxSeries,

ImportError: No module named puiseux_series_ring_element



Any ideas? I would very much like this work.

All the best,
Harry

**Comment by cswiercz** [comment]
> 10. Move to the directory abelfunctions and run sage (because I can’t manage to get sage to read a relative directory).

Try moving to your home directory or any directory ***not*** related to Abelfunctions and running Sage. I think the issue is that Sage is trying to read from the current directory and not from its package index where the actual built version of Abelfunctions lives.

### Aside

As a bit of background, Abelfunctions contains Python, Cython, and C code. The latter two need to be compiled into appropriate Python libraries (modules) before they can be imported. This compilation happens when you run 

```
$ sage setup.py install
```

or 

```
$ sage -c -p abelfunctions.spkg
```

and the compiled contents are sent to Sage's package index in the Sage application directory. ***However***, a copy of the compiled contents are not kept in the Abelfunctions source directory. This means that if you run Sage from the Abelfunctions source directory it will error because it can't find the compiled version since it will look in the current directory first. Running Sage from a different directory will force it to use the compiled version in the package index.

This is a dumb and frustrating situation and I think the installation process should also perform a `build-ext --inplace`. @chualynn : could you look into adding this?

**Comment by cswiercz** [comment]
@harrybraden Any updates? If you try installing again could you post the full output?

**Comment by harrybraden** [comment]
Dear Chris,

You want me to try from an unrelated directory. What is the correct Sage syntax for something like

from "~hwb/abelfunctions/abelfunctions”  import * 

I don’t know how to specify a directory path in Sage. Sorry for this basic question, but haven’t found it yet.

All the best,
Harry

**Comment by cswiercz** [comment]
Sorry if I wasn't clear, before. Try the following: (I'm assuming that you downloaded Abelfunctions into your home directory.)

```bash
$ cd ~/abelfunctions    # enter the Abelfunctions main directory (wherever you downloaded it)
$ sage setup.py install    # install into sage
$ cd ~    # move to a different directory...
$ sage    # ...before starting up Sage
sage: from abelfunctions import *    # any try importing Abelfunctions now
```

If you don't get out of `~/abelfunctions` then Sage will try to import the contents of `~/abelfunctions/abelfunctions` which only contains the raw source, not the compiled binaries.

If you can think of a student who would like to improve the installation process then let me know. I would be happy to chat with them.

**Comment by harrybraden** [comment]
Dear Chris,

my student is sitting exams at the moment. I have tried a number of times to install. The method you suggest gives the error message below. The other method gets further, but tell me where to start from. Here is the output of the standard install.

Thanks!
Harry

~/abelfunctions:$ sage setup.py install
running install
running build
running build_py
creating build
creating build/lib.macosx-10.9-x86_64-2.7
creating build/lib.macosx-10.9-x86_64-2.7/abelfunctions
copying abelfunctions/__init__.py -> build/lib.macosx-10.9-x86_64-2.7/abelfunctions
copying abelfunctions/abelmap.py -> build/lib.macosx-10.9-x86_64-2.7/abelfunctions
copying abelfunctions/complex_path_factory.py -> build/lib.macosx-10.9-x86_64-2.7/abelfunctions
copying abelfunctions/differentials.py -> build/lib.macosx-10.9-x86_64-2.7/abelfunctions
copying abelfunctions/divisor.py -> build/lib.macosx-10.9-x86_64-2.7/abelfunctions
copying abelfunctions/homology.py -> build/lib.macosx-10.9-x86_64-2.7/abelfunctions
copying abelfunctions/integralbasis.py -> build/lib.macosx-10.9-x86_64-2.7/abelfunctions
copying abelfunctions/puiseux.py -> build/lib.macosx-10.9-x86_64-2.7/abelfunctions
copying abelfunctions/puiseux_series_ring.py -> build/lib.macosx-10.9-x86_64-2.7/abelfunctions
copying abelfunctions/riemann_constant_vector.py -> build/lib.macosx-10.9-x86_64-2.7/abelfunctions
copying abelfunctions/riemann_surface.py -> build/lib.macosx-10.9-x86_64-2.7/abelfunctions
copying abelfunctions/riemann_surface_path.py -> build/lib.macosx-10.9-x86_64-2.7/abelfunctions
copying abelfunctions/riemann_surface_path_factory.py -> build/lib.macosx-10.9-x86_64-2.7/abelfunctions
copying abelfunctions/singularities.py -> build/lib.macosx-10.9-x86_64-2.7/abelfunctions
copying abelfunctions/skeleton.py -> build/lib.macosx-10.9-x86_64-2.7/abelfunctions
copying abelfunctions/version.py -> build/lib.macosx-10.9-x86_64-2.7/abelfunctions
copying abelfunctions/ypath_factory.py -> build/lib.macosx-10.9-x86_64-2.7/abelfunctions
creating build/lib.macosx-10.9-x86_64-2.7/abelfunctions/riemann_theta
copying abelfunctions/riemann_theta/__init__.py -> build/lib.macosx-10.9-x86_64-2.7/abelfunctions/riemann_theta
creating build/lib.macosx-10.9-x86_64-2.7/abelfunctions/utilities
copying abelfunctions/utilities/__init__.py -> build/lib.macosx-10.9-x86_64-2.7/abelfunctions/utilities
copying abelfunctions/utilities/permutations.py -> build/lib.macosx-10.9-x86_64-2.7/abelfunctions/utilities
running build_ext
cythoning abelfunctions/complex_path.pyx to abelfunctions/complex_path.c
building 'abelfunctions.complex_path' extension
C compiler: gcc -fno-strict-aliasing -I/Applications/SageMath-7.5.1.app/Contents/Resources/sage/local/var/tmp/sage/build/python2-2.7.13.p0/include -DNDEBUG -g -fwrapv -O3 -Wall -Wno-unused

creating build/temp.macosx-10.9-x86_64-2.7
creating build/temp.macosx-10.9-x86_64-2.7/abelfunctions
compile options: '-I/Applications/SageMath-7.5.1.app/Contents/Resources/sage/local/include -I/Applications/SageMath-7.5.1.app/Contents/Resources/sage/local/include/python2.7 -I/Applications/SageMath-7.5.1.app/Contents/Resources/sage/local/lib/python2.7/site-packages/numpy/core/include -I/Applications/SageMath-7.5.1.app/Contents/Resources/sage/local/lib/python2.7/site-packages -I/Applications/SageMath-7.5.1.app/Contents/Resources/sage/local/lib/python2.7/site-packages/sage/ext -I/Applications/SageMath-7.5.1.app/Contents/Resources/sage/local/lib/python2.7/site-packages/numpy/core/include -I/Applications/SageMath-7.5.1.app/Contents/Resources/sage/local/include/python2.7 -c'
extra options: '-std=c99 -w'
Warning: Can't read registry to find the necessary compiler setting
Make sure that Python modules _winreg, win32api or win32con are installed.
gcc: abelfunctions/complex_path.c
In file included from /Applications/SageMath-7.5.1.app/Contents/Resources/sage/local/lib/gcc/x86_64-apple-darwin15.6.0/5.4.0/include-fixed/syslimits.h:7:0,
                 from /Applications/SageMath-7.5.1.app/Contents/Resources/sage/local/lib/gcc/x86_64-apple-darwin15.6.0/5.4.0/include-fixed/limits.h:34,
                 from /Applications/SageMath-7.5.1.app/Contents/Resources/sage/local/include/python2.7/Python.h:19,
                 from abelfunctions/complex_path.c:4:
/Applications/SageMath-7.5.1.app/Contents/Resources/sage/local/lib/gcc/x86_64-apple-darwin15.6.0/5.4.0/include-fixed/limits.h:168:61: fatal error: limits.h: No such file or directory
compilation terminated.
In file included from /Applications/SageMath-7.5.1.app/Contents/Resources/sage/local/lib/gcc/x86_64-apple-darwin15.6.0/5.4.0/include-fixed/syslimits.h:7:0,
                 from /Applications/SageMath-7.5.1.app/Contents/Resources/sage/local/lib/gcc/x86_64-apple-darwin15.6.0/5.4.0/include-fixed/limits.h:34,
                 from /Applications/SageMath-7.5.1.app/Contents/Resources/sage/local/include/python2.7/Python.h:19,
                 from abelfunctions/complex_path.c:4:
/Applications/SageMath-7.5.1.app/Contents/Resources/sage/local/lib/gcc/x86_64-apple-darwin15.6.0/5.4.0/include-fixed/limits.h:168:61: fatal error: limits.h: No such file or directory
compilation terminated.
error: Command "gcc -fno-strict-aliasing -I/Applications/SageMath-7.5.1.app/Contents/Resources/sage/local/var/tmp/sage/build/python2-2.7.13.p0/include -DNDEBUG -g -fwrapv -O3 -Wall -Wno-unused -I/Applications/SageMath-7.5.1.app/Contents/Resources/sage/local/include -I/Applications/SageMath-7.5.1.app/Contents/Resources/sage/local/include/python2.7 -I/Applications/SageMath-7.5.1.app/Contents/Resources/sage/local/lib/python2.7/site-packages/numpy/core/include -I/Applications/SageMath-7.5.1.app/Contents/Resources/sage/local/lib/python2.7/site-packages -I/Applications/SageMath-7.5.1.app/Contents/Resources/sage/local/lib/python2.7/site-packages/sage/ext -I/Applications/SageMath-7.5.1.app/Contents/Resources/sage/local/lib/python2.7/site-packages/numpy/core/include -I/Applications/SageMath-7.5.1.app/Contents/Resources/sage/local/include/python2.7 -c abelfunctions/complex_path.c -o build/temp.macosx-10.9-x86_64-2.7/abelfunctions/complex_path.o -std=c99 -w" failed with exit status 1

**Comment by cswiercz** [comment]
> /Applications/SageMath-7.5.1.app/Contents/Resources/sage/local/lib/gcc/x86_64-apple-darwin15.6.0/5.4.0/include-fixed/limits.h:168:61: fatal error: limits.h: No such file or directory
compilation terminated.

I looked up this error on [sage-devel](https://groups.google.com/forum/#!forum/sage-devel) and based on some responses this might be due to not having XCode, or a most recent version of XCode, installed. You can download XCode from the [Apple App Store](https://itunes.apple.com/us/app/xcode/id497799835?mt=12#)

Some comments:
* XCode is a standard piece of development software so I completely overlooked adding it as a dependency for Abelfunctions. I will amend the documentation.
* That being said, you said that the alternate installation method using the Sage SPKG version of Ablelfunctions exited cleanly. Can you try the following:

  ```
  $ cd ~
  $ sage
  sage: import abelfunctions
  ```

  Since you successfully installed it using this method then it should be a part of your Sage, anyway, even if installation method 1 didn't work.

**Comment by harrybraden** [comment]
Dear Chris,

here is the (I think successful?) compilation by your other method. But what should I next do running sage?

All the best,
Harry


tar cjf abelfunctions.spkg abelfunctions/abelfunctions/.

sage -p -c abelfunctions.spkg
abelfunctions
====================================================
Extracting package /Users/hwb/abelfunctions.spkg
-rw-r--r--  1 hwb  1200627079  249150 May  9 17:46 /Users/hwb/abelfunctions.spkg
Finished extraction
/Applications/SageMath-7.5.1.app/Contents/Resources/sage/build/bin/sage-spkg: line 628: cd: src: No such file or directory
No patch files found in ../patches
****************************************************
Host system:
Darwin maths-self-022.lan 15.6.0 Darwin Kernel Version 15.6.0: Thu Jun 23 18:25:34 PDT 2016; root:xnu-3248.60.10~1/RELEASE_X86_64 x86_64
****************************************************
C compiler: gcc
C compiler version:
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/Applications/SageMath-7.5.1.app/Contents/Resources/sage/local/libexec/gcc/x86_64-apple-darwin15.6.0/5.4.0/lto-wrapper
Target: x86_64-apple-darwin15.6.0
Configured with: ../src/configure --prefix=/Applications/SageMath-7.5.1.app/Contents/Resources/sage/local --with-local-prefix=/Applications/SageMath-7.5.1.app/Contents/Resources/sage/local --with-gmp=/Applications/SageMath-7.5.1.app/Contents/Resources/sage/local --with-mpfr=/Applications/SageMath-7.5.1.app/Contents/Resources/sage/local --with-mpc=/Applications/SageMath-7.5.1.app/Contents/Resources/sage/local --with-system-zlib --disable-multilib --disable-nls --enable-languages=c,c++,fortran --disable-libitm --with-build-config=bootstrap-debug --without-isl --without-cloog
Thread model: posix
gcc version 5.4.0 (GCC)
****************************************************

real	0m0.008s
user	0m0.003s
sys	0m0.004s
Successfully installed abelfunctions
Package abelfunctions has no test suite.
Deleting temporary build directory
/Applications/SageMath-7.5.1.app/Contents/Resources/sage/local/var/tmp/sage/build/abelfunctions
Finished installing abelfunctions.spkg

Warning: it might be needed to update the Sage library before
installed packages work: you should run 'make' from $SAGE_ROOT
before running Sage.

**Comment by cswiercz** [comment]
Change back to your home directory or any directory that is ***not*** the Abelfunctions directory and run Sage:

```
$ cd ~
$ sage
sage: import abelfunctions
```

Sorry, I'm not as well-versed in Sage's preferred package installations. I found a setup that works well for me but haven't tested many other environments. I swear that the package itself, once installed, works well. :)

**Comment by cswiercz** [comment]
I also see that you're using the "App" version of Sage. I have never used this before and, instead, used the command line version of Sage so I don't know what sort of possible issues this may bring up.

I'm at work but can try installing Abelfunctions using the App version of Sage later this evening.

**Comment by harrybraden** [comment]
Dear Chris,

Is there a directory issue. The tar has abelfunctions.spkg in my top directory. I will look at Xcode as well.

H.

sage
┌────────────────────────────────────────────────────────────────────┐
│ SageMath version 7.5.1, Release Date: 2017-01-15                   │
│ Type "notebook()" for the browser-based notebook interface.        │
│ Type "help()" for help.                                            │
└────────────────────────────────────────────────────────────────────┘
Forcing sage-location, probably because a new package was installed.
Updating various hardcoded paths...
(Please wait at most a few minutes.)
DO NOT INTERRUPT THIS.
Done updating paths.
sage: import abelfunctions
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-1-22e60731dc36> in <module>()
----> 1 import abelfunctions

ImportError: No module named abelfunctions

**Comment by harrybraden** [comment]
Dear Chris,

I am running sage via the command line. 

H.

**Comment by cswiercz** [comment]
> I am running sage via the command line.

I was referring to the fact that you have the "App" version of Sage `/Applications/SageMath-7.5.1.app` which, although it has the usual command line interface, might have a different environment setup than the non-app version. As I mentioned, I'm not familiar with this distribution of Sage.

**Comment by harrybraden** [comment]
Dear Chris,

I am updating Xcode. I will let you experiment with the app.

I’ll let you know if Xcode works, otherwise I am stymied.

Thanks again,
Harry

**Comment by cswiercz** [comment]
Sounds good. As for your attempt [here](https://github.com/abelfunctions/abelfunctions/issues/142#issuecomment-300230914) I'm a bit confused as well.

**Comment by cswiercz** [comment]
In the meantime, try using a different installation of Sage. (Specifically, the non-app version. I've had success with this in the past.)

1. Go to [this page](http://www.mirrorservice.org/sites/www.sagemath.org/osx/intel/index.html)
2. Download the most recent version of the non-app version Sage compatible with out version of OSX.
    * El Capitan = 10.11
    * Sierra = 10.12
    * the app version will have "`.app`" in the filename. For example, `sage-7.6-OSX_10.11.6-x86_64.app.dmg`
3. Extract and install into your, say, home directory: `~/SageMath`
4. Run at least once
  ```
  $ ~/SageMath/sage
  ```
5. Install Abelfunctions
    * If you've installed/updated XCode then try the first method. Otherwise, try the second SPKG method
6. Run
  ```
  $ cd ~
  $ ~/SageMath/sage
  sage: import abelfunctions
  ```

If all of this doesn't work then I recommend posting on [sage-devel](https://groups.google.com/forum/#!forum/sage-devel). Include a link to this issue thread so that people can see what we've done so far. Perhaps there is something about Sage that I'm missing out on.

**Comment by harrybraden** [comment]
Dear Chris,

you can close this issue. I updated Xcode and the ordinary install worked. Thanks.

A question. The example code that calculates the period matrix and abel for the curve y^3+… was VERY slow compared to Maple’s algcurves. How quick is it for you, or is Puiseux the
problem? It was about 4 minutes for my machine.

All the best,
Harry

**Comment by markopuza** [comment]
Hello Chris,  [ @cswiercz @rparini ]

I am Harry's student and I have been trying to install abelfunctions cleanly for quite a while now. I'm running macOS Sierra, have the newest version of Xcode and Sage 8.1. I am using the .app version of Sage, but my guess is that problem is not here, as Harry seems to had made it work somehow. The issue also persists with older versions of Sage.

I have tried:
```
cd abelfunctions
sage setup.py install
```
which fails with:
```
running install
running build
running build_py
running build_ext
skipping 'abelfunctions/complex_path.c' Cython extension (up-to-date)
skipping 'abelfunctions/riemann_surface_path.c' Cython extension (up-to-date)
skipping 'abelfunctions/puiseux_series_ring_element.c' Cython extension (up-to-date)
building 'abelfunctions.puiseux_series_ring_element' extension
C compiler: gcc -fno-strict-aliasing -g -O2 -DNDEBUG -g -fwrapv -O3 -Wall -Wno-unused

compile options: '-I/Applications/SageMath-8.1.app/Contents/Resources/sage/local/include -I/Applications/SageMath-8.1.app/Contents/Resources/sage/local/include/python2.7 -I/Applications/SageMath-8.1.app/Contents/Resources/sage/local/lib/python2.7/site-packages/numpy/core/include -I/Applications/SageMath-8.1.app/Contents/Resources/sage/local/lib/python2.7/site-packages -I/Applications/SageMath-8.1.app/Contents/Resources/sage/local/lib/python2.7/site-packages/sage/ext -I/Applications/SageMath-8.1.app/Contents/Resources/sage/local/lib/python2.7/site-packages/numpy/core/include -I/Applications/SageMath-8.1.app/Contents/Resources/sage/local/include/python2.7 -c'
extra options: '-std=c99 -w'
gcc: abelfunctions/puiseux_series_ring_element.c
abelfunctions/puiseux_series_ring_element.c:526:30: fatal error: cython_metaclass.h: No such file or directory
compilation terminated.
abelfunctions/puiseux_series_ring_element.c:526:30: fatal error: cython_metaclass.h: No such file or directory
compilation terminated.
error: Command "gcc -fno-strict-aliasing -g -O2 -DNDEBUG -g -fwrapv -O3 -Wall -Wno-unused -I/Applications/SageMath-8.1.app/Contents/Resources/sage/local/include -I/Applications/SageMath-8.1.app/Contents/Resources/sage/local/include/python2.7 -I/Applications/SageMath-8.1.app/Contents/Resources/sage/local/lib/python2.7/site-packages/numpy/core/include -I/Applications/SageMath-8.1.app/Contents/Resources/sage/local/lib/python2.7/site-packages -I/Applications/SageMath-8.1.app/Contents/Resources/sage/local/lib/python2.7/site-packages/sage/ext -I/Applications/SageMath-8.1.app/Contents/Resources/sage/local/lib/python2.7/site-packages/numpy/core/include -I/Applications/SageMath-8.1.app/Contents/Resources/sage/local/include/python2.7 -c abelfunctions/puiseux_series_ring_element.c -o build/temp.macosx-10.9-x86_64-2.7/abelfunctions/puiseux_series_ring_element.o -std=c99 -w" failed with exit status 1
```
Trying the second method
```
cd ..
tar cjf abelfunctions.spkg abelfunctions
sage -p -c abelfunctions.spkg
```
looks succesfull...
```
abelfunctions
====================================================
Extracting package /Users/markopuza/Desktop/abelfunctions/abelfunctions.spkg
-rw-r--r--  1 markopuza  staff  4616636 Jan  3 20:41 /Users/markopuza/Desktop/abelfunctions/abelfunctions.spkg
Finished extraction
/Applications/SageMath-8.1.app/Contents/Resources/sage/build/bin/sage-spkg: line 646: cd: src: No such file or directory
No patch files found in ../patches
****************************************************
Host system:
Darwin Markos-MacBook-Pro.local 16.7.0 Darwin Kernel Version 16.7.0: Mon Nov 13 21:56:25 PST 2017; root:xnu-3789.72.11~1/RELEASE_X86_64 x86_64
****************************************************
C compiler: gcc
C compiler version:
Using built-in specs.
COLLECT_GCC=/Applications/SageMath-8.1.app/Contents/Resources/sage/local/bin/gcc
COLLECT_LTO_WRAPPER=/Applications/SageMath-8.1.app/Contents/Resources/sage/local/libexec/gcc/x86_64-apple-darwin15.6.0/5.4.0/lto-wrapper
Target: x86_64-apple-darwin15.6.0
Configured with: ../src/configure --prefix=/Applications/SageMath-8.1.app/Contents/Resources/sage/local --with-local-prefix=/Applications/SageMath-8.1.app/Contents/Resources/sage/local --with-gmp=/Applications/SageMath-8.1.app/Contents/Resources/sage/local --with-mpfr=/Applications/SageMath-8.1.app/Contents/Resources/sage/local --with-mpc=/Applications/SageMath-8.1.app/Contents/Resources/sage/local --with-system-zlib --disable-multilib --disable-nls --enable-languages=c,c++,fortran --disable-libitm --with-build-config=bootstrap-debug --without-isl --without-cloog
Thread model: posix
gcc version 5.4.0 (GCC)
****************************************************

real	0m0.007s
user	0m0.003s
sys	0m0.002s
Successfully installed abelfunctions
Package abelfunctions has no test suite.
Deleting temporary build directory
/Applications/SageMath-8.1.app/Contents/Resources/sage/local/var/tmp/sage/build/abelfunctions
Finished installing abelfunctions.spkg

Warning: it might be needed to update the Sage library before
installed packages work: you should run 'make' from $SAGE_ROOT
before running Sage.
```
but I am unable to import the package:
```
cd
sage 
sage: import abelfunctions
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-1-22e60731dc36> in <module>()
----> 1 import abelfunctions

ImportError: No module named abelfunctions
```
and same when sage is run as a notebook, or in ```sage -python``` mode. I have also tried to run ```make``` in the root directory of sage with no improvement.

Any help would be much appreciated, should it be even only a description of some workflow that will allow me to run abelfunctions in some way.

Best, 
Marko

**Comment by rparini** [comment]
Hi Marko,
Thanks for reporting this.  I've just tested this myself and the problem is with Sage 8.1 not having `cython_metaclass.h`.  Hopefully we can find a fix for this but in the meantime Sage 8.0 should work if you are able to downgrade.

Since this is different to Harry's issue I'll leave this one closed and I've opened a new issue here: #154

Best wishes,
Robert

**Comment by markopuza** [comment]
Hi Robert,
Thanks for the quick reply. There must be something I am missing, as I get the same failure with Sage 8.0:

```
$ /Applications/SageMath-8.0.app/sage setup.py install
running install
running build
running build_py
running build_ext
skipping 'abelfunctions/complex_path.c' Cython extension (up-to-date)
skipping 'abelfunctions/riemann_surface_path.c' Cython extension (up-to-date)
skipping 'abelfunctions/puiseux_series_ring_element.c' Cython extension (up-to-date)
building 'abelfunctions.puiseux_series_ring_element' extension
C compiler: gcc -fno-strict-aliasing -I/Applications/SageMath-8.0.app/Contents/Resources/sage/local/var/tmp/sage/build/python2-2.7.13.p1/include -DNDEBUG -g -fwrapv -O3 -Wall -Wno-unused

compile options: '-I/Applications/SageMath-8.0.app/Contents/Resources/sage/local/include -I/Applications/SageMath-8.0.app/Contents/Resources/sage/local/include/python2.7 -I/Applications/SageMath-8.0.app/Contents/Resources/sage/local/lib/python2.7/site-packages/numpy/core/include -I/Applications/SageMath-8.0.app/Contents/Resources/sage/local/lib/python2.7/site-packages -I/Applications/SageMath-8.0.app/Contents/Resources/sage/local/lib/python2.7/site-packages/sage/ext -I/Applications/SageMath-8.0.app/Contents/Resources/sage/local/lib/python2.7/site-packages/numpy/core/include -I/Applications/SageMath-8.0.app/Contents/Resources/sage/local/include/python2.7 -c'
extra options: '-std=c99 -w'
Warning: Can't read registry to find the necessary compiler setting
Make sure that Python modules _winreg, win32api or win32con are installed.
gcc: abelfunctions/puiseux_series_ring_element.c
abelfunctions/puiseux_series_ring_element.c:526:30: fatal error: cython_metaclass.h: No such file or directory
compilation terminated.
abelfunctions/puiseux_series_ring_element.c:526:30: fatal error: cython_metaclass.h: No such file or directory
compilation terminated.
error: Command "gcc -fno-strict-aliasing -I/Applications/SageMath-8.0.app/Contents/Resources/sage/local/var/tmp/sage/build/python2-2.7.13.p1/include -DNDEBUG -g -fwrapv -O3 -Wall -Wno-unused -I/Applications/SageMath-8.0.app/Contents/Resources/sage/local/include -I/Applications/SageMath-8.0.app/Contents/Resources/sage/local/include/python2.7 -I/Applications/SageMath-8.0.app/Contents/Resources/sage/local/lib/python2.7/site-packages/numpy/core/include -I/Applications/SageMath-8.0.app/Contents/Resources/sage/local/lib/python2.7/site-packages -I/Applications/SageMath-8.0.app/Contents/Resources/sage/local/lib/python2.7/site-packages/sage/ext -I/Applications/SageMath-8.0.app/Contents/Resources/sage/local/lib/python2.7/site-packages/numpy/core/include -I/Applications/SageMath-8.0.app/Contents/Resources/sage/local/include/python2.7 -c abelfunctions/puiseux_series_ring_element.c -o build/temp.macosx-10.9-x86_64-2.7/abelfunctions/puiseux_series_ring_element.o -std=c99 -w" failed with exit status 1
```

**Comment by markopuza** [comment]
Update:

I managed to run abelfunctions on a different computer. Whatever the issue is, it is probably sage-level. I don't have anaconda on a different computer - some sources claimed that might have helped?...

Still, even though the ``` setup.py ``` installation executed cleanly, it gave ``` ImportError: No module named puiseux_series_ring_element ``` when trying to import the module, as in Harry's case.

Reinstalling via the ```spkg``` method finally worked.

Thanks again!

**Comment by tolsonMDA** [comment]
I looked into this issue briefly this morning, and it looks like the `cython_metaclass.h` header has been moved in sage 8.1 from `sage/math/` to `sage/cypthon/` in [this commit](https://github.com/sagemath/sage/commit/ed0d88c9c42f1f7d10eebba7c4a29e620709072d). The failing import appears to be generated automatically by cython (there's no reference to it in the `.pxd` file), so I suppose there is a configuration path somewhere that needs to be updated to use this new location.
